torch_xla/csrc/runtime/computation_client.h

```diff
@@ -107,13 +149,41 @@ class ComputationClient {
 
     const xla::ProgramShape& program_shape() const { return program_shape_; }
 
+    const torch::lazy::hash_t& hash() const { return hash_; }
+
+    int parameters_size() const override {
+      return program_shape().parameters_size();
+    }
+
+    const std::vector<torch::lazy::Shape>& parameter_shapes() const override {
+      XLA_ERROR() << "Unimplemented";
```

Comment:
are these `XLA_ERROR` because they are not currently being used?

Reply:
Correct. There was not even any logic before to convert the XLA parameter shapes to lazy shapes in the previous implementation. We just returned a default-initialized object, which seems more dangerous to me.

Upstream defines this as a pure virtual function, but I could not find any usage.